consensus/parlia/parlia.go

```diff
@@ -884,6 +898,35 @@ func (p *Parlia) prepareValidators(header *types.Header) error {
 	return nil
 }
 
+func (p *Parlia) prepareTurnTerm(chain consensus.ChainHeaderReader, header *types.Header) error {
+	if header.Number.Uint64()%p.config.Epoch != 0 ||
+		!p.chainConfig.IsBohr(header.Number, header.Time) {
+		return nil
+	}
+
+	parent := chain.GetHeaderByHash(header.ParentHash)
+	if parent == nil {
+		return errors.New("parent not found")
+	}
+
+	if p.chainConfig.IsBohr(parent.Number, parent.Time) {
+		turnTerm, err := p.getTurnTerm(parent)
+		if err != nil {
+			return err
+		}
+		if turnTerm == nil {
+			return errors.New("unexpected error when getTurnTerm")
+		}
+		header.Extra = append(header.Extra, byte(int(turnTerm.Int64())))
+		log.Debug("prepareTurnTerm", "turnTerm", turnTerm.Int64())
+	} else {
+		log.Debug("prepareTurnTerm", "turnTerm", "defaultTurnTerm")
```

Comment:
This line cannot be reached, because the previous `IsBohr` check has returned.

Reply:
> This line cannot be reached, because the previous `IsBohr` check has returned.

differernt: IsBohr(header... vs IsBohr(parent...  
It will be reached, if the epoch block is the first Bohr Block